lib/SILOptimizer/Utils/ConstExpr.cpp

```diff
@@ -587,17 +598,76 @@ ConstExprFunctionState::computeWellKnownCallResult(ApplyInst *apply,
            conventions.getNumIndirectSILResults() == 0 &&
            conventions.getNumParameters() == 4 && "unexpected signature");
     auto literal = getConstantValue(apply->getOperand(1));
-    if (literal.getKind() != SymbolicValue::String)
-      break;
+    if (literal.getKind() != SymbolicValue::String) {
+      return evaluator.getUnknown((SILInstruction *)apply,
+                                  UnknownReason::Default);
+    }
```

Comment:
@marcrasi I added support for String.+= and also String.==. In addition to that I also changed the breaks here to return unknown. I wasn't sure why there was a break here? But, it seemed to me that it was perhaps because there are two functions with "makeUTF8" semantics attribute and the evaluator is handling just one of them. But, it turns out that the other one is also identical (in terms of semantics) and is redundant and is probably unused.